site/src/api/api.ts

```diff
@@ -779,3 +779,10 @@ export const getTemplateVersionLogs = async (
   )
   return response.data
 }
+
+export const updateWorkspaceVersion = async (
+  workspace: TypesGen.Workspace,
+): Promise<TypesGen.WorkspaceBuild> => {
+  const template = await getTemplate(workspace.template_id)
```

Comment:
Do we want to handle the error for `getTemplate`?

Reply:
It is already handled. If it fails, it will throw an error that will be passed to the `Alert` component. But I guess, you were talking about handling errors in a more granular way. Since these operations are consecutive, one by one, the first one that fails will throw the error, so only one error will happen at a time. But even in case we could have two errors happening at the same time in this operation, I would ask: As a user are those two errors helpful? Or would be good enough to just show a single error like "Update template version failed"?  IMO, as a user, usually don't need to know what is happening under the hood so I would avoid showing multiple errors which can make me confused. Makes sense?

Reply:
Sure, totally makes sense. I just see a blank screen with a toast when that error is thrown. Maybe down the road and outside of the PR, we can have a cuter error state, like even an empty table.

Reply:
Ah, it should not be a toast, should be an alert banner, not VERY different tho. I agree we could have better error components when there is a main loading data error.